### ISSUE TITLE
import parser corrupts the import table

### ISSUE THREAD

**Issue by thewhitegoatcb** [Qwokka/WAIL#3]
Hello, been having an issue
```
let parser = new WailParser(binary);
parser.getGlobalIndex(0); //just so the import section gets parsed
parser.parse();
let output = parser.write(); //this produces the corrupt output
```

It seems to output 2 bytes short at the end of the table omitting the kind and type byte of the last entry (shifting the next section into the 2 bytes).
Happened on quite a big binary but was able to reproduce on a small demo one

[data.zip](https://github.com/Qwokka/WAIL/files/8965090/data.zip)

**Comment by thewhitegoatcb** [comment]
figured it out, commitBytes was missing after the kind and type is rad on the last iteration.
will make a pull request later on

**Comment by Qwokka** [comment]
Thanks for catching this @thewhitegoatcb !

If you'd prefer to make the pull request I'll gladly merge it. Otherwise I'll make the change as you described it.

Jack

**Comment by thewhitegoatcb** [comment]
Go ahead,
I think export parsing might have a similar issue haven't tested myself.
I suggest moving commitBytes to the end of the for loop and call one before the for loop starts(if any data is rad), also this might allow easier implementation of the modifications so you can perform the logic in the same loop and discard/replace the bytes if wanted.

**Comment by Qwokka** [comment]
This should be fixed with 08281538a30f14c370132f545509d3060e2bbb71 just as you described.

I'm not sure if the issue applies to exports since WAIL rewrites each export to account for possible changes to the function or global tables. That has the unintended effect of not requiring a call to commitBytes() after each entry. Of course I missed this issue for so long that I could definitely be missing something else.

Thanks again for taking the time on this! Feel free to reopen if I missed anything.

Jack
